if statement returns "invalid memory address or nil pointer dereference" error

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->

After adding an extra condition `(and (eq .name "argocd-applicationset-controller") (eq .port 7000))` when running `goss -g test.yaml --vars vars.yaml validate` I get this error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x77b9e0]
```

**How To Reproduce**
<!-- Please provide a minimal reproducible example
https://stackoverflow.com/help/minimal-reproducible-example -->

vars.yaml
```
tcp_services:
  - name: argo-rollouts-dashboard
    namespace:  argo-rollouts
    port: 3100
  - name: argo-rollouts-metrics
    namespace:  argo-rollouts
    port: 8090
  - name: argocd-applicationset-controller
    namespace:  argocd
    port: 7000
  - name: argocd-applicationset-controller
    namespace:  argocd
    port: 8080
```

test.yaml
```
command:
  # Ensure TCP services are not reachable by any pod, except Traefik and some other core services
  {{ range index .Vars "tcp_services" }}
  "kubectl -n demo exec test-network-policy -ti -- /bin/nc -vz -w 1 {{.name}}.{{.namespace}} {{.port}}":
    timeout: 30000 # in milliseconds
    {{ if or (eq .name "lb-traefik") (eq .name "kube-dns") (eq .name "prometheus-adapter") (and (eq .name "argocd-applicationset-controller") (eq .port 7000)) }}
    exit-status: 0
    stderr:
    - "/{{.name}}.{{.namespace}} .*:{{.port}}.* open/"
    {{ else }}
    exit-status: 1
    stderr:
    - "/{{.name}}.{{.namespace}} .*:{{.port}}.* Connection timed out/"
    {{ end }}
  {{end}}
```

**Expected Behavior**
<!-- A clear and concise description of what you expected to happen. -->

I want to be able to test for 2 conditions at the same time in a complex if statement.

**Actual Behavior**
<!-- A clear and concise description of what actually happened. -->

I got a segfault error:

```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x77b9e0]

goroutine 65 [running]:
github.com/goss-org/goss/system.(*DefHTTP).Status(0xc0004ee0d0)
        /home/travis/gopath/src/github.com/goss-org/goss/system/http.go:184 +0x40
github.com/goss-org/goss/resource.ValidateValue({0xc2d958, 0xc0005ca8c0}, {0xb29627, 0x6}, {0xa4aa80, 0x1018980}, {0xa51580?, 0xc0002820a8?}, _)
        /home/travis/gopath/src/github.com/goss-org/goss/resource/validate.go:135 +0x369
github.com/goss-org/goss/resource.(*HTTP).Validate(0xc0005ca8c0, 0x0?)
        /home/travis/gopath/src/github.com/goss-org/goss/resource/http.go:77 +0x305
github.com/goss-org/goss.validate.func2()
        /home/travis/gopath/src/github.com/goss-org/goss/validate.go:171 +0x91
created by github.com/goss-org/goss.validate
        /home/travis/gopath/src/github.com/goss-org/goss/validate.go:168 +0x190
```

**Environment:**
 - Version of goss: v0.3.21
 - OS/Distribution version (if applicable)
  cat /etc/debian_version
  bookworm/sid

Got it again! It's related with the `http` module:

```
$ ./goss --version
goss version v0.4.1
		
$ cat goss.yaml
http:
  # Ensure argocd is available and redirected to HTTPS
  http://argocd.mgmt.{{ .Vars.domain_name }}/:
    status: 302
    no-follow-redirects: true
    headers:
    - "Location: https://argocd"
	
$ cat vars.yaml
domain_name: #DOMAIN#
k8s_version: 1.25

$ ./goss -g goss.yaml --vars vars.yaml render
http:
  http://argocd.mgmt.<no value>/:
    status: 302
    allow-insecure: false
    no-follow-redirects: true
    headers:
    - 'Location: https://argocd'
	
$ ./goss -g goss.yaml --vars vars.yaml  validate
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x7ac440]

goroutine 69 [running]:
github.com/goss-org/goss/system.(*DefHTTP).Status(0xc00012b380)
        /home/travis/gopath/src/github.com/goss-org/goss/system/http.go:186 +0x40
github.com/goss-org/goss/resource.ValidateGomegaValue({_, _}, {_, _}, {_, _}, {_, _}, _)
        /home/travis/gopath/src/github.com/goss-org/goss/resource/validate.go:160 +0x3f6
github.com/goss-org/goss/resource.ValidateValue({_, _}, {_, _}, {_, _}, {_, _}, _)
        /home/travis/gopath/src/github.com/goss-org/goss/resource/validate.go:130 +0x195
github.com/goss-org/goss/resource.(*HTTP).Validate(0xc000460a20, 0xc000330370?)
        /home/travis/gopath/src/github.com/goss-org/goss/resource/http.go:81 +0x305
github.com/goss-org/goss.validate.func2()
        /home/travis/gopath/src/github.com/goss-org/goss/validate.go:184 +0x91
created by github.com/goss-org/goss.validate
        /home/travis/gopath/src/github.com/goss-org/goss/validate.go:181 +0x190
```

Hope that with this new info you can fix it :)

--- a/system/http.go
+++ b/system/http.go
@@ -89,6 +89,7 @@ func (u *DefHTTP) setup() error {
 		parseProxy, err := url.Parse(u.Proxy)
 
 		if err != nil {
+			u.err = err
 			return err
 		}
 
@@ -103,12 +104,15 @@ func (u *DefHTTP) setup() error {
 		// FIXME: iotutil
 		caCert, err := os.ReadFile(u.CAFile)
 		if err != nil {
+			u.err = err
 			return err
 		}
 		roots := x509.NewCertPool()
 		ok := roots.AppendCertsFromPEM(caCert)
 		if !ok {
-			return fmt.Errorf("Failed parse root certificate: %s", u.CAFile)
+			err := fmt.Errorf("Failed parse root certificate: %s", u.CAFile)
+			u.err = err
+			return err
 		}
 		tlsConfig.RootCAs = roots
 	}
@@ -116,6 +120,7 @@ func (u *DefHTTP) setup() error {
 	if u.CertFile != "" && u.KeyFile != "" {
 		cert, err := tls.LoadX509KeyPair(u.CertFile, u.KeyFile)
 		if err != nil {
+			u.err = err
 			return err
 		}
 
@@ -140,6 +145,7 @@ func (u *DefHTTP) setup() error {
 
 	req, err := http.NewRequest(u.Method, u.http, strings.NewReader(u.RequestBody))
 	if err != nil {
+		u.err = err
 		return u.err
 	}
 	req.Header = u.RequestHeader.Clone()

--- a/system/http.go
+++ b/system/http.go
@@ -83,13 +83,18 @@ func (u *DefHTTP) setup() error {
 		return u.err
 	}
 	u.loaded = true
+	if err := u.setupReal(); err != nil {
+		u.err = err
+	}
+	return u.err
 
+}
+func (u *DefHTTP) setupReal() error {
 	proxyURL := http.ProxyFromEnvironment
 	if u.Proxy != "" {
 		parseProxy, err := url.Parse(u.Proxy)
 
 		if err != nil {
-			u.err = err
 			return err
 		}
 
@@ -104,15 +109,12 @@ func (u *DefHTTP) setup() error {
 		// FIXME: iotutil
 		caCert, err := os.ReadFile(u.CAFile)
 		if err != nil {
-			u.err = err
 			return err
 		}
 		roots := x509.NewCertPool()
 		ok := roots.AppendCertsFromPEM(caCert)
 		if !ok {
-			err := fmt.Errorf("Failed parse root certificate: %s", u.CAFile)
-			u.err = err
-			return err
+			return fmt.Errorf("Failed parse root certificate: %s", u.CAFile)
 		}
 		tlsConfig.RootCAs = roots
 	}
@@ -120,7 +122,6 @@ func (u *DefHTTP) setup() error {
 	if u.CertFile != "" && u.KeyFile != "" {
 		cert, err := tls.LoadX509KeyPair(u.CertFile, u.KeyFile)
 		if err != nil {
-			u.err = err
 			return err
 		}
 
@@ -145,8 +146,7 @@ func (u *DefHTTP) setup() error {
 
 	req, err := http.NewRequest(u.Method, u.http, strings.NewReader(u.RequestBody))
 	if err != nil {
-		u.err = err
-		return u.err
+		return err
 	}
 	req.Header = u.RequestHeader.Clone()
 

